nevergrad/optimization/test_optimizerlib.py

```diff
@@ -508,23 +522,96 @@ def _ellips(x: np.ndarray) -> float:
-    metamodel_recom, default_recom = recommendations  # pylint: disable=unbalanced-tuple-unpacking
+    for baseline in ["CMA", "ECMA"]:
+        num_trials = 1 if os.environ.get("CIRCLECI", False) else 7
+        successes = 0
+        for _ in range(num_trials):
+            if successes > num_trials // 2:  # We already have enough
+                break
+
+            # Let us run the comparison.
+            recommendations: tp.List[np.ndarray] = []
+            for name in ("MetaModel", baseline if dimension > 1 else "OnePlusOne"):
+                opt = registry[name](dimension, contextual_budget, num_workers=num_workers)
+                recommendations.append(opt.minimize(_target).value)
+            metamodel_recom, default_recom = recommendations  # pylint: disable=unbalanced-tuple-unpacking
+
+            # Let us assert that MetaModel is better.
+            if not _target(default_recom) > _target(metamodel_recom):
+                continue
+
+            # With large budget, the difference should be significant.
+            if budget > 60 * dimension:
+                if not _target(default_recom) > 4.0 * _target(metamodel_recom):
+                    continue
+
+            # ... even more in the non ellipsoid case.
+            if budget > 60 * dimension and not ellipsoid:
+                if not _target(default_recom) > 7.0 * _target(metamodel_recom):
+                    continue
+            successes += 1
+        assert successes > num_trials // 2, f"Problem for beating {baseline}."
 
-    # Let us assert that MetaModel is better.
-    assert _target(default_recom) > _target(metamodel_recom)
 
-    # With large budget, the difference should be significant.
-    if budget > 60 * dimension:
-        assert _target(default_recom) > 4.0 * _target(metamodel_recom)
+@testing.suppress_nevergrad_warnings()
+@skip_win_perf  # type: ignore
+@pytest.mark.parametrize(
+    "dimension, num_workers, scale, budget, ellipsoid, baseline",
+    [
+        (d, 1, s, b, e, baseline)
+        for e in [True, False]
+        for b in [200, 500]
+        for s in [8.0]
+        for d in [7]
+        for baseline in ["MetaModel", "CMA", "ECMA"]
+    ][::5],
+)
+def test_metamodel_sqp_chaining(
+    dimension: int, num_workers: int, scale: float, budget: int, ellipsoid: bool, baseline: str
+) -> None:
+    """The test can operate on the sphere or on an elliptic funciton."""
+
+    if os.environ.get("CIRCLECI", False):  # This test is too expensive for CircleCI.
+        return
+
+    def _square(x: np.ndarray) -> float:
+        return sum((-scale + x) ** 2)
 
-    # ... even more in the non ellipsoid case.
-    if budget > 60 * dimension and not ellipsoid:
-        assert _target(default_recom) > 7.0 * _target(metamodel_recom)
+    def _ellips(x: np.ndarray) -> float:
+        return sum(((-scale + x) * (np.arange(1, dimension + 1) ** 2)) ** 2)
+
+    _target = _ellips if ellipsoid else _square
```

Comment:
no need to add `_` for local variables
also you could directly define target, with a `if`